crates/k_means/src/lib.rs

```diff
@@ -46,6 +47,7 @@ pub trait KMeans {
         });
     }
     fn finish(self: Box<Self>) -> Square;
+    fn finish_samples(self: Box<Self>) -> (Square, Square);
```

Comment:
This should not return samples. The implementation details of K-means can certainly involve some irreversible operations.

Reply:
Fixed.